diff --git a/QuagBot/src/app/components/Chat/ChatHistory.jsx b/QuagBot/src/app/components/Chat/ChatHistory.jsx
index ddc5f10..b33c276 100644
--- a/QuagBot/src/app/components/Chat/ChatHistory.jsx
+++ b/QuagBot/src/app/components/Chat/ChatHistory.jsx
@@ -1,7 +1,7 @@
 import { useEffect, useState } from "react";
 import ChatPrevious from "./ChatPrevious";
 
-function ChatHistory({ onSelectChat }) {
+function ChatHistory({ onSelectChat, activeChat }) {
   const [chats, setChats] = useState([
     {
       title: "Example Chat", content: [
@@ -66,7 +66,7 @@ function ChatHistory({ onSelectChat }) {
     <div className="max-w-50 overflow-x-hidden text-nowrap">
       <h2 className="sidebar-title">Chat History</h2>
       <div>
-        {chats.map((chat, i) => <ChatPrevious key={i} chat={chat} onSelect={onSelectChat} />)}
+        {chats.map((chat, i) => <ChatPrevious key={i} chat={chat} isActive={chat === activeChat} onSelect={onSelectChat} />)}
       </div>
     </div>
   );
diff --git a/QuagBot/src/app/components/Chat/ChatPrevious.jsx b/QuagBot/src/app/components/Chat/ChatPrevious.jsx
index b592735..886e867 100644
--- a/QuagBot/src/app/components/Chat/ChatPrevious.jsx
+++ b/QuagBot/src/app/components/Chat/ChatPrevious.jsx
@@ -1,6 +1,6 @@
 import { useEffect, useState } from "react";
 
-function ChatPrevious({ chat, onSelect }) {
+function ChatPrevious({ chat, isActive, onSelect }) {
   if (!chat) return
 
   function setAsActiveChat(chat) {
@@ -11,7 +11,7 @@ function ChatPrevious({ chat, onSelect }) {
     <div
       title={chat.title}
       onClick={() => setAsActiveChat(chat)}
-      className="text-sm w-full rounded-xl cursor-pointer py-1.25 px-5 text-left hover:bg-gray-300 "
+      className={`text-sm w-full rounded-xl cursor-pointer py-1.25 px-5 text-left hover:bg-gray-300 ${isActive ? "bg-gray-300" : ""}`}
     >
       {chat.title}
     </div>
diff --git a/QuagBot/src/app/page.js b/QuagBot/src/app/page.js
index 4f2c281..c9a0d3f 100644
--- a/QuagBot/src/app/page.js
+++ b/QuagBot/src/app/page.js
@@ -19,6 +19,7 @@ function Home() {
     const [textSize, setTextSize] = useState('18px');
     const [font, setFont] = useState('Arial, sans-serif');
     const [fontColor, setFontColor] = useState('#000000');
+    const [activeChat, setActiveChat] = useState(null);
     const pendingReply = useRef(null);
 
     function sendTextToBot(text) {
@@ -48,12 +49,13 @@ function Home() {
         }
         setIsLoading(false);
         setMessages([...chat.content]);
+        setActiveChat(chat);
     }
 
     return (
         <PageWrapper>
             <Sidebar title="" side="left">
-                <ChatHistory onSelectChat={loadChat} />
+                <ChatHistory onSelectChat={loadChat} activeChat={activeChat} />
             </Sidebar>
 
             <ChatWindow
